atc/exec/retry_error_step.go

```diff
@@ -48,18 +50,18 @@ func (step RetryErrorStep) Run(ctx context.Context, state RunState) error {
 }
 
 func (step RetryErrorStep) toRetry(logger lager.Logger, err error) bool {
-	switch err.(type) {
-	case transport.WorkerMissingError, transport.WorkerUnreachableError:
+	var transportErr *transport.WorkerMissingError
+	var unreachable *transport.WorkerUnreachableError
+	var netOpErr *net.OpError
+	re := regexp.MustCompile(`worker .+ disappeared`)
+	if ok := errors.As(err, &transportErr) || errors.As(err, &unreachable) || errors.As(err, &netOpErr); ok {
```

Comment:
Rather than handling `*net.OpError`, I think it's better to handle `net.Error` (an interface that `*net.OpError` implements).

Additionally, I ran into a `*url.Error` during my testing, which apparently usually wraps a `net.Error` but not always. So, could you add a case for `*url.Error` as well?